### PR TITLE
[WFLY-15009] - SecurityContextImpl.login incorrectly assumes authenti…

### PR DESCRIPTION
**Pull request by adamecp**
Jira issue: [https://issues.redhat.com/browse/ELYWEB-133](https://issues.redhat.com/browse/ELYWEB-133)
Jira WFLY issue [https://issues.redhat.com/browse/ELY-2185:](https://issues.redhat.com/browse/ELY-2185)

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/SecuredServlet.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A simple secured HTTP servlet.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+@WebServlet("/secured")
+@ServletSecurity(httpMethodConstraints = { @HttpMethodConstraint(value = "GET", rolesAllowed = { "Users" }) })
+public class SecuredServlet extends HttpServlet {
```
Comment (darranl) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/SecuredServlet.java:39]: Did this need to be cloned?  Was it not possible to reuse an existing copy of this class?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/MyDummyTokenHandler.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.jboss.as.test.integration.security.securitycontext;
```
Comment (darranl) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/MyDummyTokenHandler.java:1]: Missing copyright header

**Review by darranl — CHANGES_REQUESTED** [review]
As a change to the WildFly codebase this needs to use a WFLY Jira issue and not the issue from an external project.

As this is just testing an alternative authentication flow would it not be possible to enhance an existing test case instead of adding a completely new setup / deploy / test / clean up for a single scenario?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/SecuredServlet.java (hunk 1 above)
Comment (adamecp) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/SecuredServlet.java:39]: on the firts time, I tried modify existing test , but I had some problems with it so then I write new one.

Comment (adamecp) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/securitycontext/SecuredServlet.java]: I have removed SecuredServlet.java  and reuse existing SecuredServlet.java

**Comment by adamecp** [comment]
@darranl Should I  make some other changes?

**Comment by darranl** [comment]
@adamecp it looks like the test case is failing on CI.

**Comment by adamecp** [comment]
@darranl Can I replan CI tests and run them again? Maybe problems were due to environment fails. I have  just tried test on local machine and it is ok.

**Comment by darranl** [comment]
This one is a re-run I triggered this morning before I realised the same test was failing in all of them:
    https://ci.wildfly.org/buildConfiguration/WF_PullRequest_BootableJarLinuxJdk11/265168?

**Comment by adamecp** [comment]
I know where is a problem. I'm rewriting it.

**Comment by adamecp** [comment]
Now it si ok. I will rename one method

**Comment by adamecp** [comment]
@darranl Now It is complete. All tests passed, so pleas merge it

**Review by darranl — CHANGES_REQUESTED** [review]
Looking good, one small change remaining is the year in the copyright header for new files should be 2021.  Any files forked from elsewhere can keep their existing header.

**Comment by adamecp** [comment]
ok. I have changed year in SecurityContextImplLoginTestCase.java

**Review by darranl — APPROVED** [review]

**Comment by adamecp** [comment]
@darranl Thanks for approve. Could you merge it, please?

**Pull request merged**